src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -112,7 +112,9 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
 
                     var  projectInstance = _projectInstanceFactory(
                         projectReferenceFullPath,
-                        null, // Platform negotiation requires an evaluation with no global properties first
+                        new Dictionary<string, string>(){
+                            {"PlatformNegotiationOuterBuild", "true"},
```

Comment:
This doesn't feel right, maybe it's my understanding.

What exactly does `PlatformNegotiationOuterBuild` here mean? Is it similar to "outer builds" like building a `net472;net6.0` project and how it has individual "inner builds" of `net472` and `net6.0`? Or does "outer build" here mean something unique to platform negotiation?